modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java

```diff
@@ -431,9 +431,16 @@ public void processOpts() {
 
         if (!interfaceOnly) {
             if (SPRING_BOOT.equals(library)) {
+                if (useSwaggerUI) {
+                    supportingFiles.add(new SupportingFile("swagger-ui.mustache", "src/main/resources/static", "swagger-ui.html"));
```

Comment:
Why limit this when not interface only?

Reply:
interfaceOnly (as I understand) means the user does not care about the generated spring-boot project. So no spring-boot runtime, no swagger-ui. The same goes for the spring-cloud library. Did I miss understood the interfaceOnly parameter?

Reply:
I use interfaceOnly because we implement our own controllers - mostly because we didn't want to modify the generated controller code each time we regenerated from the API specification.

We annotate the interface with `@Controller` instead of the implementation class. 

We do still use SpringDoc - and this seems like an alternative or supplement to that.

Reply:
But then you provide the dependencies and configuration for springDoc by yourself - following the `apiFirst` approach?

Reply:
Correct - which we would need to do anyway as we're using Bazel for our build system and dependency management as opposed to Maven (although it uses Maven to actually fetch jars)

Unrelated - but maybe long term, I can work on adding Bazel support here.

Reply:
I still think this should be moved outside the interface check, but otherwise PR looks good to me

Reply:
Sorry, I don't get it - What would you do with the generated swagger-ui.html without the necessary spring-boot infrastructure? The Spring Boot Main class and configurations have always been skipped in **interfaceOnly** mode.

Reply:
Configure it manually - you can still have Spring Boot set up even if it's not generated from here and the generated HTML could still be used